module/zfs/dsl_scan.c

```diff
@@ -141,9 +142,9 @@ dsl_scan_init(dsl_pool_t *dp, uint64_t txg)
 		 */
 		if (err == EOVERFLOW) {
```

Comment:
it locks you into the current *size* of the dsl_scan_phys_t, but adding the new flag should be OK.